File: backend/src/main/java/com/example/backend/model/Venta.java
package com.example.backend.model;
import com.example.backend.enums.FormaPago;
import java.time.LocalDateTime;
import jakarta.persistence.*;
import java.math.BigDecimal;
import java.util.List;
import lombok.*;

@Entity
@Table(name = "venta")
@Getter @Setter @ToString(exclude = "detalles") @Builder @NoArgsConstructor @AllArgsConstructor
public class Venta {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private BigDecimal total;

    @Enumerated(EnumType.STRING)
    @Column(name = "forma_pago")
    private FormaPago formaPago; 

    private String observacion;

    @Column(name = "nombre_cliente")
    private String nombreCliente;

    @Column(name = "fecha_venta")
    private LocalDateTime fechaVenta;

    @OneToMany(mappedBy = "venta", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<DetalleVenta> detalles;
}

File: frontend-clean/src/app/core/models/venta.ts
export interface VentaRequest {
  nombreCliente: string;
  total: number;
  formaPago: string;
  observacion?: string;
  detalles: {
    idProducto: number;
    cantidad: number;
  }[];
}

export interface VentaDetalle {
  id: number;
  idProducto: number;
  nombreProducto: string;
  cantidad: number;
  precioUnitario: number;
  subtotal: number;
}

export interface VentaView {
  id: number;
  total: number;
  formaPago: string;
  observacion?: string;
  nombreCliente: string;
  fechaVenta: string;
  detalles: VentaDetalle[];
}